From 8daab38dfc1fe4d3df9fb5fc18610b942d5cc3b2 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 12 May 2024 14:28:19 +0200
Subject: [PATCH] Simplify coredump memcpy in mwl_fwcmd_get_fw_core_dump

Simplify coredump memcpy in mwl_fwcmd_get_fw_core_dump. Instead of doing
fragile address additions, just access the buffer member in pcmd and
reference the pointer.

This fix a compilation warning in 64Bit system:

3608 | (const void *)((u32)pcmd +
| ^

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 hif/fwcmd.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/hif/fwcmd.c
+++ b/hif/fwcmd.c
@@ -3624,11 +3624,7 @@ int mwl_fwcmd_get_fw_core_dump(struct ie
 	core_dump->context = pcmd->cmd_data.coredump.context;
 	core_dump->size_kb = pcmd->cmd_data.coredump.size_kb;
 	core_dump->flags = pcmd->cmd_data.coredump.flags;
-	memcpy(buff,
-	       (const void *)((u32)pcmd +
-	       sizeof(struct hostcmd_cmd_get_fw_core_dump) -
-	       sizeof(struct hostcmd_cmd_get_fw_core_dump_)),
-	       MAX_CORE_DUMP_BUFFER);
+	memcpy(buff, pcmd->buffer, MAX_CORE_DUMP_BUFFER);
 
 	mutex_unlock(&priv->fwcmd_mutex);
 
